clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/RawResourceProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Coveo, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model;
+
+import com.netflix.spinnaker.clouddriver.documentation.Empty;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A rawResourceProvider is an interface for the application to retrieve {@link RawResource}
+ * objects. The interface provides a common contract for which one or many providers can be queried
+ * for their knowledge of raw resources at a given depth of specificity.
+ */
+public interface RawResourceProvider {
```

Comment:
If these interfaces are intended for implementation by cloud providers, they belong in `clouddriver-api`.

As such, the interfaces will require additional documentation (each method, nested interface, etc), to implement `SpinnakerExtensionPoint`, and be annotated with `@Alpha`.

Reply:
clouddriver-api is pretty empty. Is it a new thing?

Reply:
It is a new thing, yes.

Reply:
I had trouble moving these interfaces to `clouddriver-api`. For example, the NamerRegistry utility isn't available. What's the best way to do this? Only provide the function definitions in `clouddriver-api` and implement the NamerRegistry call in the Kubernetes provider?